### PR TITLE
feat(client): added progress widget

### LINKED ISSUE #25: UI - Progress widget
image

**Is your feature request related to a problem? Please describe.**
As someone learning how to code, I want to see how far I am generally across the languages/technologies I know.

**Describe the solution you'd like**
There should be a widget on the dashboard that displays all paths and the progress the user is through them.

**Additional context**
May want to use a canvas to draw the arc.
https://www.figma.com/file/MfOc0a8Ac9Zoh9RppoEuiD/CM-UI?node-id=2%3A0

**Data**
This will need dummy data for now. It should follow the format of:
```ts
interface PathProgress {
  path: Path // Still being worked on by @raysercast1 
  progress: number; // 0 - 1
}

// Dummy interface until it's available from #5 
interface Path {
  id: string;
  name: string;
  icon: string;
  description: string;
}

<ProgressWidget /> // fetches its own data (mocked for now)
```

### PR DESCRIPTION
**Pull request by ahmedfaaid**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Fixed issues (PRs without this will not be accepted)
- Fixes #25 

## Changes
- Created a Progress Widget that displays all a user's paths and their progress through them

## PR Checklist
- [x] Does your PR contain relevant tests that pass?
- [x] Is your code linted? `yarn lint` or `yarn lint:fix`
- [x] Does your code build correctly in ALL projects? `yarn build`

### CODE REVIEW

File: packages/client/src/components/ProgressWidget/ProgressWidget.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        <div className="inline-block mx-1">
+          <svg width="48" height="48" className="cursor-pointer">
+            <g>
+              <circle
+                stroke="#EAE7F6"
+                cx={24}
+                cy={24}
+                r={23}
+                strokeWidth={4}
+                strokeDasharray="10,10"
```
Comment (tristanMatthias) [inline packages/client/src/components/ProgressWidget/ProgressWidget.tsx:48]: Let's change this to `4,7`. Bit of a nicer dash

File: packages/client/src/components/ProgressWidget/ProgressWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                fill="none"
+              />
+            </g>
+            <Icon
```
Comment (tristanMatthias) [inline packages/client/src/components/ProgressWidget/ProgressWidget.tsx:52]: Can we get some margin around this? It feels too cramped up against the circle.

File: packages/client/src/components/ProgressWidget/ProgressPath.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            className="svg-rotate transform -rotate-90"
+          />
+        </g>
+        <Icon
+          icon="x"
+          size="small"
+          className="text-primary-200"
+        />
+      </svg>
+      <p className="text-center text-sm text-secondary-500 font-sans font-extrabold ">{`${progress}%`}</p>
```
Comment (tristanMatthias) [inline packages/client/src/components/ProgressWidget/ProgressPath.tsx:61]: Can we also get `letter-spacing: 1px` on this? I think tailwind has a class for it

**Comment by ahmedfaaid** [comment]
@tristanMatthias I'll get right on it

File: packages/client/src/components/ProgressWidget/ProgressWidget.tsx (hunk 2 above)
Comment (ahmedfaaid) [reply packages/client/src/components/ProgressWidget/ProgressWidget.tsx:52]: Would it be okay if I make the circle a little bigger?

**Comment by ahmedfaaid** [comment]
Hey @tristanMatthias I rebased before pushing so I had to force push. Hope that is okay.

**Pull request merged**